pkg/providers/snow/snow.go

```diff
@@ -118,14 +118,23 @@ func CAPIObjects(ctx context.Context, clusterSpec *cluster.Spec, kubeClient kube
 		return nil, nil, err
 	}
 
-	workersSpec, err = templater.ObjectsToYaml(workersObjs...)
+	workersSpec, err = templater.ObjectsToYaml(kubernetesToRuntimeObjects(workersObjs)...)
 	if err != nil {
 		return nil, nil, err
 	}
 
 	return controlPlaneSpec, workersSpec, nil
 }
 
+func kubernetesToRuntimeObjects(objs []kubernetes.Object) []runtime.Object {
```

Comment:
Can you explain the difference between kubernetes and runtime objects?

Reply:
Good question. `runtime.Object` is the basic kubernetes interface for api objects. `client.Object` is the interface used by the `controller-runtime` `Client`, which is a superset of `runtime.Object`. Which is the reason why we can't use `runtime.Object` with the controller code.

`kubernetes.Object` is the interface we created to represent api objects in our eks-a code, in particular when dealing with share code between the controller and CLI. `kubernetes.Object` should always implement `client.Object`

We could use `client.Client` directly in the CLI code as well but that seemed like a dangerous commitment. By having our own interface, we can add more stuff to it without necessarily having to update the signature of all methods in those pieces of shared code.